webhook/webhook.go

```diff
@@ -164,6 +167,15 @@ func (wh *Webhook) Run(stop <-chan struct{}) error {
 				if err != nil {
 					return nil, err
 				}
+				certData, err := x509.ParseCertificate(cert.Certificate[0])
+				if err != nil {
+					logger.Info("failed to parse certificate: " + err.Error())
+				}
+				oneWeek := 168 * time.Hour
+				if certData.NotAfter.Sub(time.Now()) < oneWeek {
+					wh.Client.CoreV1().Secrets(system.Namespace()).Delete(secret.Name, &metav1.DeleteOptions{})
```

Comment:
The secret isn't created by the reconciler, so this will not work as intended.

I'd change here: https://github.com/knative/pkg/blob/1cc7b7152937735a29c9d6fdaf3b244684aa850e/webhook/certificates/certificates.go#L70

Reply:
That makes sense. I was actually able to delete the secret so many times that it just didn't come back and couldn't figure out how to trigger it to be created again. I'll look into this and make updates.

Reply:
I just wrote a surprisingly similar check for auto-TLS here: https://github.com/mattmoor/http01-solver/blob/4f7f1732d30512a74cbeb5d88341dfeb26c546ba/pkg/reconciler/certificate/certificate.go#L70

dereferencing that here: https://github.com/mattmoor/http01-solver/blob/4f7f1732d30512a74cbeb5d88341dfeb26c546ba/pkg/reconciler/certificate/resources/secret.go#L62-L66

The logic should be set up in a way so that if you just add this check things should automatically refresh the cert, we just only do it for malformed certs today.